analyzers/maven/maven.go

```diff
@@ -73,19 +75,46 @@ func (a *Analyzer) Discover(dir string) ([]module.Module, error) {
 
 			dir := filepath.Dir(path)
 			submodules, err := a.Maven.Modules(dir)
-			if err != nil {
-				log.Logger.Debugf("Modules err: %#v %#v", err.Error(), err)
-				return err
-			}
+			if len(submodules) == 0 {
+				// Failed to parse modules via calling into `mvn` or this is a single-module build
+				// Fallback to parsing the pom.xml and skipping the underlying file tree assuming they are subprojects
+				if err != nil {
+					log.Logger.Debugf("Unable to parse modules through `mvn`: %#v %#v", err.Error(), err)
+				}
+				var parsedPom maven.Manifest
+				pomParseError := files.ReadXML(&parsedPom, path, "pom.xml")
+
+				groupId := parsedPom.GroupID
+				if groupId == "" {
+					groupId = parsedPom.Parent.GroupID
+				}
+
+				if parsedPom.Name == "" || groupId == "" || parsedPom.ArtifactID == "" {
+					pomParseError = errors.Errorf("invalid manifest at %s/pom.xml", path)
+				}
+
+				if pomParseError != nil {
```

Comment:
This error check must occur before `pom` is used. Using return values of function calls with non-nil errors is UB.